File: java/anagrams.java
public class Solution {
    public ArrayList<String> anagrams(String[] strs) {
        // Start typing your Java solution below
        // DO NOT write main() function
        ArrayList<String> ans = new ArrayList<String>();
        HashMap<String, ArrayList<String>> map = new HashMap<String, ArrayList<String>>();
        for(String e : strs ){
            char[] chare1 = e.toCharArray();
            Arrays.sort(chare1);
            String chare = new String(chare1);
            if(map.keySet().contains(chare)){
                map.get(chare).add(e);
            }else{
                ArrayList<String> al = new ArrayList<String>();
                al.add(e);
                map.put(chare, al);
            }
        }
        for(String e : map.keySet()){
            if(map.get(e).size()>1){
                 ans.addAll(map.get(e));
            }
        }

        return ans;
    }

}

File: cpp/Anagrams.cpp
class Solution {
public:
    vector<string> anagrams(vector<string> &strs) {
        unordered_map<string, vector<string> > sortedToString;
        for(auto it=strs.begin(); it!=strs.end(); it++) {
            string copy = *it;
            sort(copy.begin(), copy.end());
            vector<string> v = sortedToString[copy];
            v.push_back(*it);
            sortedToString[copy] = v;
        }
        vector<string> res;
        for(auto it=sortedToString.begin(); it!=sortedToString.end(); it++) {
            if (it->second.size()>1) {
                res.insert(res.end(), it->second.begin(), it->second.end());
            }
        }
        return res;
    }
};
